File: backend/src/main/java/com/airport/kiosk/model/Booking.java
package com.airport.kiosk.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "bookings")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Booking {
    
    @Id
    @Column(name = "booking_id", length = 50)
    private String bookingId;
    
    @Column(name = "passenger_name", nullable = false, length = 255)
    private String passengerName;
    
    @Column(name = "passport_number", length = 50)
    private String passportNumber;
    
    @Column(name = "email", length = 255)
    private String email;
    
    @Column(name = "phone", length = 50)
    private String phone;
    
    @Column(name = "flight_id", nullable = false, length = 50)
    private String flightId;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "booking_status", nullable = false)
    private BookingStatus bookingStatus = BookingStatus.CONFIRMED;
    
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
    
    public enum BookingStatus {
        CONFIRMED, CHECKED_IN, CANCELLED
    }
}

File: frontend/src/types/booking.ts
import type { Flight } from './flight';

export interface Booking {
  bookingId: string;
  passengerName: string;
  passportNumber?: string;
  email?: string;
  phone?: string;
  flightId: string;
  bookingStatus: 'CONFIRMED' | 'CHECKED_IN' | 'CANCELLED';
  createdAt: string;
  updatedAt: string;
}

export interface BookingSearchRequest {
  bookingReference?: string;
  passportNumber?: string;
}

export interface BookingSearchResponse {
  booking: Booking;
  flight: Flight;
}
